lib/charms/opensearch/v0/opensearch_base_charm.py

```diff
@@ -542,23 +575,39 @@ def _can_service_start(self) -> bool:
 
         return True
 
-    def _initialize_admin_user(self):
-        """Change default password of Admin user."""
-        hashed_pwd, pwd = generate_hashed_password()
-        self.secrets.put(Scope.APP, "admin_password", pwd)
+    def _put_admin_user(self, pwd: Optional[str] = None):
```

Comment:
**suggestion** making this parametric with the username if we plan to extend this?

Reply:
This i would prefer to leave for later - when we introduce new users. As it won't be enough to just add the `username`, but also the `backend_roles` and potentially other attributes we don't yet know about.